src/sentry/integrations/github/webhook.py

```diff
@@ -90,7 +91,23 @@ def __call__(self, event: Mapping[str, Any], host: str | None = None) -> None:
                 provider=f"integrations:{self.provider}",
                 external_id=str(event["repository"]["id"]),
             )
-            for repo in repos:
+
+            if not repos.exists():
+                provider = get_integration_repository_provider(integration)
+
+                config = {
+                    "integration_id": integration.id,
+                    "external_id": str(event["repository"]["id"]),
+                    "identifier": event.get("repository", {}).get("full_name", None),
+                }
+
+                for org in list(orgs.values()):
+                    if features.has("organizations:auto-repo-linking", org):
+                        provider.create_repository(config, org)
+
+                repos = repos.all()
+
+            for repo in repos.exclude(status=ObjectStatus.HIDDEN):
```

Comment:
Just to confirm if a repo is is`HIDDEN` but a push/pull event comes in, it shouldnt be set to `ACTIVE`?

Reply:
that would be correct. we wouldn't set it to `ACTIVE`